### ISSUE TITLE
Cannot start payara after updating weld to 2.3 and guava to 18 

### ISSUE THREAD

**Issue by drupalspring** [payara/Payara#437]
I download payara (4.1.153 Full Java EE) and want to update the weld to 2.3 and guava to 18 because of the following bugs:

https://github.com/payara/Payara/issues/376
https://github.com/payara/Payara/issues/431 

My update method is simply to replace `guava.jar` and `weld-osgi-bundle.jar` inside `modules` folders by jars with latest version . For Weld , I have already replaced it with `Weld OSGi bundle` which is downloaded from http://search.maven.org/#artifactdetails%7Corg.jboss.weld%7Cweld-osgi-bundle%7C2.3.0.CR2%7Cjar . I can successfully update weld and guava in glassfish 4.1 in this way but payara throws the following errors during the server start:

> Sep 14, 2015 12:35:07 PM com.sun.enterprise.glassfish.bootstrap.osgi.BundleProvisioner createBundleProvisioner
INFO: Create bundle provisioner class = class com.sun.enterprise.glassfish.bootstrap.osgi.BundleProvisioner.
Sep 14, 2015 12:35:07 PM com.sun.enterprise.glassfish.bootstrap.osgi.BundleProvisioner$DefaultCustomizer getLocations
WARNING: Skipping entry  because it is not an absolute URI.
Sep 14, 2015 12:35:07 PM com.sun.enterprise.glassfish.bootstrap.osgi.BundleProvisioner$DefaultCustomizer getLocations
WARNING: Skipping entry  because it is not an absolute URI.
Exception in thread "main" java.lang.reflect.InvocationTargetException
  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:483)
  at com.sun.enterprise.glassfish.bootstrap.GlassFishMain.main(GlassFishMain.java:97)
  at com.sun.enterprise.glassfish.bootstrap.ASMain.main(ASMain.java:54)
Caused by: com.sun.enterprise.module.ResolveError: Failed to start OSGiModuleImpl:: Bundle = [org.glassfish.main.batch.glassfish-batch-connector [95]], State = [NEW]
  at org.jvnet.hk2.osgiadapter.OSGiModuleImpl.start(OSGiModuleImpl.java:218)
  at org.jvnet.hk2.osgiadapter.OsgiPopulatorPostProcessor$1.loadClass(OsgiPopulatorPostProcessor.java:77)
  at org.jvnet.hk2.config.ConfigModel.getProxyType(ConfigModel.java:167)
  at org.jvnet.hk2.config.Dom.getImplementationClass(Dom.java:1016)
  at org.jvnet.hk2.internal.JrServiceLocatorProxy.registerServiceClass(JrServiceLocatorProxy.java:211)
  at org.jvnet.hk2.internal.ServiceLocatorImpl.reifyDescriptor(ServiceLocatorImpl.java:420)
  at org.jvnet.hk2.internal.ServiceLocatorImpl.reifyDescriptor(ServiceLocatorImpl.java:430)
  at sun.reflect.GeneratedMethodAccessor20.invoke(Unknown Source)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:483)
  at org.jvnet.hk2.internal.JrServiceLocatorProxy.invoke(JrServiceLocatorProxy.java:69)
  at org.glassfish.hk2.api.ServiceLocator_$$_jvst307_0.reifyDescriptor(ServiceLocator_$$_jvst307_0.java)
  at org.jvnet.hk2.config.Dom.register(Dom.java:208)
  at org.jvnet.hk2.config.ConfigParser.handleElement(ConfigParser.java:252)
  at org.jvnet.hk2.config.ConfigParser.handleElement(ConfigParser.java:190)
  at org.jvnet.hk2.config.ConfigParser.handleElement(ConfigParser.java:231)
  at org.jvnet.hk2.config.ConfigParser.handleElement(ConfigParser.java:238)
  at org.jvnet.hk2.config.ConfigParser.handleElement(ConfigParser.java:238)
  at org.jvnet.hk2.config.ConfigParser.handleElement(ConfigParser.java:190)
  at org.jvnet.hk2.config.ConfigParser.parse(ConfigParser.java:100)
  at org.jvnet.hk2.config.ConfigParser.parse(ConfigParser.java:94)
  at org.glassfish.config.support.DomainXml.parseDomainXml(DomainXml.java:259)
  at org.glassfish.config.support.DomainXml.run(DomainXml.java:117)
  at org.jvnet.hk2.config.ConfigurationPopulator.populateConfig(ConfigurationPopulator.java:58)
  at org.glassfish.hk2.bootstrap.HK2Populator.populateConfig(HK2Populator.java:106)
  at com.sun.enterprise.module.common_impl.AbstractModulesRegistryImpl.populateConfig(AbstractModulesRegistryImpl.java:211)
  at com.sun.enterprise.module.bootstrap.Main.createServiceLocator(Main.java:273)
  at org.jvnet.hk2.osgiadapter.HK2Main.createServiceLocator(HK2Main.java:120)
  at com.sun.enterprise.glassfish.bootstrap.osgi.EmbeddedOSGiGlassFishRuntime.newGlassFish(EmbeddedOSGiGlassFishRuntime.java:95)
  at com.sun.enterprise.glassfish.bootstrap.GlassFishRuntimeDecorator.newGlassFish(GlassFishRuntimeDecorator.java:68)
  at com.sun.enterprise.glassfish.bootstrap.osgi.OSGiGlassFishRuntime.newGlassFish(OSGiGlassFishRuntime.java:91)
  at com.sun.enterprise.glassfish.bootstrap.GlassFishMain$Launcher.launch(GlassFishMain.java:113)
  ... 6 more
Caused by: org.osgi.framework.BundleException: Unresolved constraint in bundle org.glassfish.main.batch.glassfish-batch-connector [95]: Unable to resolve 95.0: missing requirement [95.0] osgi.wiring.package; (osgi.wiring.package=com.ibm.jbatch.container.context.impl) [caused by: Unable to resolve 238.0: missing requirement [238.0] osgi.wiring.package; (osgi.wiring.package=org.glassfish.weld) [caused by: Unable to resolve 293.0: missing requirement [293.0] osgi.wiring.package; (&(osgi.wiring.package=org.jboss.weld)(version>=2.2.0)(!(version>=3.0.0)))]]
  at org.apache.felix.framework.Felix.resolveBundleRevision(Felix.java:3974)
  at org.apache.felix.framework.Felix.startBundle(Felix.java:2037)
  at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:955)
  at org.jvnet.hk2.osgiadapter.OSGiModuleImpl.start(OSGiModuleImpl.java:210)
>  ... 37 more

**Comment by smillidge** [comment]
Hi this is an OSGI dependencies problem. I'm not sure why this is failing to bind as the requirement is for weld >= 2.2 and < 3.0 which should be satisfied. It maybe that the server will need to be built against the latest weld as the org.glassfish.weld package is unable to find the org.jboss.weld package of the correct version.

You could check the OSGI metadata in the org.glassfish.weld bundle to see if it corresponds to the metadata in your weld osgi bundle.

**Comment by smillidge** [comment]
There may be a problem with the manifest for Weld 2.3.0.Final I had to add the imports
```shell
org.jboss.classfilewriter.annotations;version="1.1",
org.jboss.classfilewriter.constpool;version="1.1",
org.jboss.classfilewriter.attributes;version="1.1"
```

You also have upgrade JBoss Logging to 3.3.0.Final in the modules directory
And add jboss-classfilewriter-1.1.1.Final.jar into the modules directory

**Comment by drupalspring** [comment]
@smillidge  : Can you solve the problem once you upgrade JBoss Logging to 3.3.0.Final and add jboss-classfilewriter-1.1.1.Final.jar in modules directory and patch manifest in  Weld 2.3.0.Final ? I do the same but it has the following error during server startup , any idea:

2015-10-15T14:31:51.424+0800|Severe: org.osgi.framework.BundleException: Unresolved constraint in bundle org.glassfish.fighterfish.osgi-http [302]: Unable to resolve 302.0: missing requirement [302.0] osgi.wiring.package; (&(osgi.wiring.package=com.sun.enterprise.web)(version>=3.1.1)) [caused by: Unable to resolve 277.0: missing requirement [277.0] osgi.wiring.package; (&(osgi.wiring.package=com.sun.enterprise.web.connector.grizzly)(version>=4.1.0)(!(version>=5.0.0))) [caused by: Unable to resolve 275.0: missing requirement [275.0] osgi.wiring.package; (&(osgi.wiring.package=org.glassfish.web.loader)(version>=4.1.0)(!(version>=5.0.0))) [caused by: Unable to resolve 273.0: missing requirement [273.0] osgi.wiring.package; (osgi.wiring.package=org.jboss.weld.interceptor.proxy) [caused by: Unable to resolve 287.0: missing requirement [287.0] osgi.wiring.host; (&(osgi.wiring.host=org.jboss.weld.osgi-bundle)(bundle-version>=0.0.0))]]]]
	at org.apache.felix.framework.Felix.resolveBundleRevision(Felix.java:3974)
	at org.apache.felix.framework.Felix.startBundle(Felix.java:2037)
	at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1291)
	at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:304)
	at java.lang.Thread.run(Thread.java:745)
2015-10-15T14:31:51.533+0800|Info: ERROR: Bundle org.glassfish.fighterfish.osgi-web-container [307] Error starting file:/Users/ken/Downloads/glassfish4.1.1-20150928/glassfish/modules/autostart/osgi-web-container.jar (org.osgi.framework.BundleException: Unresolved constraint in bundle org.glassfish.fighterfish.osgi-web-container [307]: Unable to resolve 307.0: missing requirement [307.0] osgi.wiring.package; (&(osgi.wiring.package=com.sun.enterprise.web)(version>=4.0.0)) [caused by: Unable to resolve 277.0: missing requirement [277.0] osgi.wiring.package; (&(osgi.wiring.package=com.sun.enterprise.web.connector.grizzly)(version>=4.1.0)(!(version>=5.0.0))) [caused by: Unable to resolve 275.0: missing requirement [275.0] osgi.wiring.package; (&(osgi.wiring.package=org.glassfish.web.loader)(version>=4.1.0)(!(version>=5.0.0))) [caused by: Unable to resolve 273.0: missing requirement [273.0] osgi.wiring.package; (osgi.wiring.package=org.jboss.weld.interceptor.proxy) [caused by: Unable to resolve 287.0: missing requirement [287.0] osgi.wiring.host; (&(osgi.wiring.host=org.jboss.weld.osgi-bundle)(bundle-version>=0.0.0))]]]])
2015-10-15T14:31:51.534+0800|Severe: org.osgi.framework.BundleException: Unresolved constraint in bundle org.glassfish.fighterfish.osgi-web-container [307]: Unable to resolve 307.0: missing requirement [307.0] osgi.wiring.package; (&(osgi.wiring.package=com.sun.enterprise.web)(version>=4.0.0)) [caused by: Unable to resolve 277.0: missing requirement [277.0] osgi.wiring.package; (&(osgi.wiring.package=com.sun.enterprise.web.connector.grizzly)(version>=4.1.0)(!(version>=5.0.0))) [caused by: Unable to resolve 275.0: missing requirement [275.0] osgi.wiring.package; (&(osgi.wiring.package=org.glassfish.web.loader)(version>=4.1.0)(!(version>=5.0.0))) [caused by: Unable to resolve 273.0: missing requirement [273.0] osgi.wiring.package; (osgi.wiring.package=org.jboss.weld.interceptor.proxy) [caused by: Unable to resolve 287.0: missing requirement [287.0] osgi.wiring.host; (&(osgi.wiring.host=org.jboss.weld.osgi-bundle)(bundle-version>=0.0.0))]]]]
	at org.apache.felix.framework.Felix.resolveBundleRevision(Felix.java:3974)
	at org.apache.felix.framework.Felix.startBundle(Felix.java:2037)
	at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1291)
	at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:304)
	at java.lang.Thread.run(Thread.java:745)
2015-10-15T14:31:51.840+0800|Info: HV000001: Hibernate Validator 5.1.2.Final

**Comment by smillidge** [comment]
Fixed by #641
